### PR TITLE
Helm: add TTL configuration for schedule and workflow

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: Cwen Yin <cwenyin0@gmail.com>

### What problem does this PR solve?

 

#2922 

### What's changed and how it works?

. 

1. add TTL configuration for `schedule` and workflow` in `value.yaml` 
2. add `FinishTime` instead of `EndTime` 

### Related changes

- [x] Need to update `chaos-mesh/website`: https://github.com/chaos-mesh/website/pull/267
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/dashboard/core/workflow.go [hunk 1]
```diff
@@ -51,15 +51,16 @@ const (
...
-	Status    WorkflowStatus `json:"status,omitempty"`
-	Archived  bool           `json:"-"`
+	ID         uint           `gorm:"primary_key" json:"id"`
+	UID        string         `gorm:"index:workflow_uid" json:"uid"`
+	Namespace  string         `json:"namespace"`
+	Name       string         `json:"name"`
+	Entry      string         `json:"entry"` // the entry node name
+	CreatedAt  time.Time      `json:"created_at"`
+	FinishTime time.Time      `json:"finish_time"`
+	EndTime    string         `json:"end_time"`
```
Comment (iguoyr) [inline pkg/dashboard/core/workflow.go:61]: How about adding some comments to describe the difference between these two fields?

**Review by iguoyr — COMMENTED** [review]
Rest LGTM

File: pkg/dashboard/core/workflow.go (hunk 1 above)
Comment (cwen0) [reply pkg/dashboard/core/workflow.go:61]: Got it

Comment (cwen0) [reply pkg/dashboard/core/workflow.go]: Fixed! @iguoyr PTAL again! Thanks!

File: pkg/config/dashboard/dashboard.go [hunk 2]
```diff
@@ -75,25 +78,37 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
 }
 
 // ParsePersistTTLConfig parse PersistTTLConfig to persistTTLConfigParsed.
-func ParsePersistTTLConfig(config *PersistTTLConfig) (*ttlcontroller.TTLconfig, error) {
-	SyncPeriod, err := time.ParseDuration(config.SyncPeriod)
+func ParsePersistTTLConfig(config *PersistTTLConfig) (*ttlcontroller.TTLConfig, error) {
+	syncPeriod, err := time.ParseDuration(config.SyncPeriod)
 	if err != nil {
-		return nil, err
+		return nil, errors.WithStack(err)
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:84]: I prefer to use `errors.Wrap(err, "prase configuration sync period")` with the message about what we are doing here.

File: pkg/config/dashboard/dashboard.go [hunk 3]
```diff
@@ -75,25 +78,37 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
...
 	}
 
-	Event, err := time.ParseDuration(config.Event)
+	event, err := time.ParseDuration(config.Event)
 	if err != nil {
-		return nil, err
+		return nil, errors.WithStack(err)
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:89]: ditto, `errors.Wrap(err, "prase configuration TTL for event")`

File: pkg/config/dashboard/dashboard.go [hunk 4]
```diff
@@ -75,25 +78,37 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
...
 	}
 
-	Experiment, err := time.ParseDuration(config.Experiment)
+	experiment, err := time.ParseDuration(config.Experiment)
 	if err != nil {
-		return nil, err
+		return nil, errors.WithStack(err)
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:94]: ditto

File: pkg/config/dashboard/dashboard.go [hunk 5]
```diff
@@ -75,25 +78,37 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
...
 	}
 
-	return &ttlcontroller.TTLconfig{
-		DatabaseTTLResyncPeriod: SyncPeriod,
-		EventTTL:                Event,
-		ArchiveTTL:              Experiment,
+	schedule, err := time.ParseDuration(config.Schedule)
+	if err != nil {
+		return nil, errors.WithStack(err)
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:99]: ditto

File: pkg/config/dashboard/dashboard.go [hunk 6]
```diff
@@ -75,25 +78,37 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
...
+	}
+
+	workflow, err := time.ParseDuration(config.Workflow)
+	if err != nil {
+		return nil, errors.WithStack(err)
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:104]: ditto

**Review by STRRL — COMMENTED** [review]
rest LGTM

File: pkg/config/dashboard/dashboard.go [hunk 7]
```diff
@@ -81,27 +81,27 @@ func GetChaosDashboardEnv() (*ChaosDashboardConfig, error) {
...
 	syncPeriod, err := time.ParseDuration(config.SyncPeriod)
 	if err != nil {
-		return nil, errors.WithStack(err)
+		return nil, errors.Wrap(err, "parse configuration sync period")
 	}
 
 	event, err := time.ParseDuration(config.Event)
 	if err != nil {
-		return nil, errors.WithStack(err)
+		return nil, errors.Wrap(err, "parse configuration sync period")
```
Comment (STRRL) [inline pkg/config/dashboard/dashboard.go:89]: maybe use a different error message, like `parse configuration TTL for event`?

Or you prefer to use "parse configuration sync period" for all of them? 🙈

Comment (cwen0) [reply pkg/config/dashboard/dashboard.go:89]: Oh, sorry, I will fix it.

Comment (cwen0) [reply pkg/config/dashboard/dashboard.go]: Fixed!

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by iguoyr — APPROVED** [review]
LGTM

**Pull request merged**